Sources/StreamChat/Database/DTOs/UserDTO_Tests.swift

```diff
@@ -141,116 +141,19 @@ class UserDTO_Tests: XCTestCase {
-        
-        // Save the changed payload with the same hash
-        try database.writeSynchronously { session in
-            try session.saveUser(payload: changedPayload)
-        }
-        
-        // Load the user from the db and check the fields are correct
-        var loadedUserModel: _ChatUser<NoExtraData>? {
-            database.viewContext.user(id: userId)?.asModel()
-        }
-        
-        // Assert that properties are not changed
-        XCTAssertEqual(payload.name, loadedUserModel?.name)
-        XCTAssertEqual(payload.imageURL, loadedUserModel?.imageURL)
-        XCTAssertEqual(payload.isOnline, loadedUserModel?.isOnline)
-        XCTAssertEqual(payload.isBanned, loadedUserModel?.isBanned)
-        XCTAssertEqual(payload.role, loadedUserModel?.userRole)
-        XCTAssertEqual(payload.createdAt, loadedUserModel?.userCreatedAt)
-        XCTAssertEqual(payload.updatedAt, loadedUserModel?.userUpdatedAt)
-        XCTAssertEqual(payload.lastActiveAt, loadedUserModel?.lastActiveAt)
-        XCTAssertEqual(payload.teams, loadedUserModel?.teams)
-        
-        XCTAssertNotEqual(changedPayload.name, loadedUserModel?.name)
-        XCTAssertNotEqual(changedPayload.imageURL, loadedUserModel?.imageURL)
-        XCTAssertNotEqual(changedPayload.isOnline, loadedUserModel?.isOnline)
-        XCTAssertNotEqual(changedPayload.isBanned, loadedUserModel?.isBanned)
-        XCTAssertNotEqual(changedPayload.role, loadedUserModel?.userRole)
-        XCTAssertNotEqual(changedPayload.createdAt, loadedUserModel?.userCreatedAt)
-        XCTAssertNotEqual(changedPayload.updatedAt, loadedUserModel?.userUpdatedAt)
-        XCTAssertNotEqual(changedPayload.lastActiveAt, loadedUserModel?.lastActiveAt)
-        
-        let newPayload: UserPayload<NoExtraData> = .dummy(userId: userId)
-        
-        // Save the changed payload with the same hash
-        try database.writeSynchronously { session in
-            try session.saveUser(payload: newPayload)
-        }
-        
-        // Assert that properties are changed
-        // since the `newPayload` has is different
-        XCTAssertEqual(newPayload.name, loadedUserModel?.name)
-        XCTAssertEqual(newPayload.imageURL, loadedUserModel?.imageURL)
-        XCTAssertEqual(newPayload.isOnline, loadedUserModel?.isOnline)
-        XCTAssertEqual(newPayload.isBanned, loadedUserModel?.isBanned)
-        XCTAssertEqual(newPayload.role, loadedUserModel?.userRole)
-        XCTAssertEqual(newPayload.createdAt, loadedUserModel?.userCreatedAt)
-        XCTAssertEqual(newPayload.updatedAt, loadedUserModel?.userUpdatedAt)
-        XCTAssertEqual(newPayload.lastActiveAt, loadedUserModel?.lastActiveAt)
-        XCTAssertEqual(newPayload.teams, loadedUserModel?.teams)
+
+        // Act: Save payload again
+        let user = try database.viewContext.saveUser(payload: payload)
+
+        // Assert: DTO should not contain any changes
+        XCTAssertFalse(user.hasPersistentChangedValues)
```

Comment:
I like how we have almost 1k tests written in a certain way and you went "YOLO I'll do `Arrange / Act / Assert`" 😅